javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,6 +56,24 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
+        if (size == 0) { return empty(); }
+
+        Object[] array = (iterable instanceof Collection<?>) ? ((Collection<?>) iterable).toArray()
+                                                             : asArray(iterable.iterator(), size);
+        return ofAll(array, size, 0);
+    }
+
+    static <T> BitMappedTrie<T> ofAll(Object[] array, int size, int depthShift) {
+        if (size == 0) { return empty(); }
+
+        int shift = depthShift;
+        for (; array.length > branchingFactor(); shift += BRANCHING_BASE) {
```

Comment:
Because `branchingFactor()` is referential transparent and has no arguments it is effectively constant. I know that the call will be optimized away at runtime but I would write it as constant:

```java
static int BRANCHING_FACTOR = 1 << BRANCHING_BASE;
```

It isn't final because of your branching base tests, right?

We should make these fields/methods `private static final` and use [a hack](http://stackoverflow.com/questions/3301635/change-private-static-final-field-using-java-reflection) for the tests instead.

The tests could overwrite both fields, the BRANCHING_FACTOR does not need to be a method!

We should not weaken code because of unit tests!